### DISCUSSION TITLE
[Show and Tell] RewardMe - Send crypto assets to anyone using only receiver's email or username

### DISCUSSION THREAD

**Discussion by Romsters** [zkSync-Community-Hub/zksync-developers#190] [Show and Tell]
### Project Name

RewardMe

### Repo Link

https://github.com/romsters/rewardme

### Demo Link (Optional)

https://rewardme-app-kxknkq7kaq-lm.a.run.app

### Technologies Used

Solidity, Rust, React, wagmi, mui, Google & Github API

### Describe your project. What does it do? Why did you create it? Any interesting challenges you faced?

The project allows you to send crypto assets (`ETH`, `ERC20`, `NFT`) as rewards to anyone without knowing receiver's account address. All you need to have is receiver's `GitHub` username or `Google` account email.
Receiver is able to claim the rewards by proving ownership of `GitHub` or `Google` account (by using Sign in with `GitHub` or Sign in with `Google` flow).

You can connect with Metamask or WalletConnect. You can use it from your mobile browser as well.
One of my goals was to support both L1  and L2 in terms of development, deployment  and testing, and make it work end-to-end with both L1 and zkSync Era.

For now the project is deployed to the following networks:
- zkSync Era Mainnet (L2)
- zkSync Era Goerli Testnet (L2)
- Sepolia (L1)
- zkSync Era Local (L2) - local node, you can use it if you have your local node and block explorer running locally, you also need to [deploy the contracts](https://github.com/Romsters/rewardme/blob/main/README.md#-deploying-contracts).
So make sure you use one of these ^.

No sensitive data is stored anywhere, username or email is hashed before it's stored in SC.

Claiming flow is: Get token from Google / GitHub -> Exchange it for a signed app token -> Smart contract verifies the app token -> The smart contract sends you your rewards to the address you choose.

<img width="385" alt="Screenshot 2023-10-29 at 04 03 13" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/8478134/c0673a2f-2348-4d0e-927e-19fc7a50d346">
<img width="291" alt="Screenshot 2023-10-29 at 04 07 24" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/8478134/81c6e329-6a75-4ff2-b8a6-4b7942973e8f">

### What specific feedback are you looking for?

Try it out, let me know what do you think.

### How can the community help or contribute to your project?

Try it, test it, check out the source code and feel free to reuse it!

### Additional Information (Optional)

The project is a prototype, don't use it or it's parts on production if you are not 100% sure.

**Comment by mztacat** [comment]
will work on this in a bit

**Comment by Jeslyn8933-888** [comment]
Being in the moment is being mindful,

**Comment by Cryptodelhi** [comment]
done testing or mainnet

**Comment by jamra13** [comment]
Very nice idea, congrats!

**Comment by cosmossphere** [comment]
This is really cool

**Comment by GaodiXC** [comment]
cool

**Comment by foodara90** [comment]
we all community this is awosone

**Comment by Crypto2049** [comment]
> ### Project Name
> RewardMe
> 
> ### Repo Link
> https://github.com/romsters/rewardme
> 
> ### Demo Link (Optional)
> https://rewardme-app-kxknkq7kaq-lm.a.run.app
> 
> ### Technologies Used
> Solidity, Rust, React, wagmi, mui, Google & Github API
> 
> ### Describe your project. What does it do? Why did you create it? Any interesting challenges you faced?
> The project allows you to send crypto assets (`ETH`, `ERC20`, `NFT`) as rewards to anyone without knowing receiver's account address. All you need to have is receiver's `GitHub` username or `Google` account email. Receiver is able to claim the rewards by proving ownership of `GitHub` or `Google` account (by using Sign in with `GitHub` or Sign in with `Google` flow).
> 
> You can connect with Metamask or WalletConnect. You can use it from your mobile browser as well. One of my goals was to support both L1 and L2 in terms of development, deployment and testing, and make it work end-to-end with both L1 and zkSync Era.
> 
> For now the project is deployed to the following networks:
> 
> * zkSync Era Mainnet (L2)
> * zkSync Era Goerli Testnet (L2)
> * Sepolia (L1)
> * zkSync Era Local (L2) - local node, you can use it if you have your local node and block explorer running locally, you also need to [deploy the contracts](https://github.com/Romsters/rewardme/blob/main/README.md#-deploying-contracts).
>   So make sure you use one of these ^.
> 
> No sensitive data is stored anywhere, username or email is hashed before it's stored in SC.
> 
> Claiming flow is: Get token from Google / GitHub -> Exchange it for a signed app token -> Smart contract verifies the app token -> The smart contract sends you your rewards to the address you choose.
> 
> <img alt="Screenshot 2023-10-29 at 04 03 13" width="385" src="https://user-images.githubusercontent.com/8478134/278847228-c0673a2f-2348-4d0e-927e-19fc7a50d346.png"> <img alt="Screenshot 2023-10-29 at 04 07 24" width="291" src="https://user-images.githubusercontent.com/8478134/278847320-81c6e329-6a75-4ff2-b8a6-4b7942973e8f.png">
> ### What specific feedback are you looking for?
> Try it out, let me know what do you think.
> 
> ### How can the community help or contribute to your project?
> Try it, test it, check out the source code and feel free to reuse it!
> 
> ### Additional Information (Optional)
> The project is a prototype, don't use it or it's parts on production if you are not 100% sure.

**Reply by TOLANI767** [reply]
good job guys

**Comment by 0xnth** [comment]
great idea

**Comment by shankt66** [comment]
This is greatest idea! Those people got there wallet drained can reicieved there rewards by email or there new wallet.

**Comment by jaguard2021** [comment]
Dude, it's awesome, useful tool

**Comment by hbriese** [comment]
Great idea but any unclaimed funds can be rugged

**Comment by Dollyerls** [comment]
If you plan to batch send employee rewards, how should you do it? I recommend adding a batch function; and after batch sending, the list status (sent successfully/received/rejected and return to sender) can be added. You can also consider releasing the rewards in batches or linearly.

**Comment by mccoy90** [comment]
This is nice, will work on it

**Comment by telixgoldens** [comment]
Great idea. Is the app running already now. Would love to try it out

**Comment by Vito1115** [comment]
good stuff fren...

**Comment by infamousgee** [comment]
Good idea. What is the reward going to be based off of. It will be interesting to see

**Comment by Emily1232020** [comment]
very cool

**Comment by Criptosik** [comment]
> ### название проекта
> Вознаградить меня
> 
> ### Ссылка на репо
> https://github.com/romsters/rewardme
> 
> ### Ссылка на демо (необязательно)
> https://rewardme-app-kxknkq7kaq-lm.a.run.app
> 
> ### Используемые технологии
> Solidity, Rust, React, wagmi, mui, API Google и Github
> 
> ### Опишите свой проект. Что оно делает? Зачем вы это создали? Были ли какие-нибудь интересные задачи, с которыми вы столкнулись?
> Проект позволяет отправлять криптоактивы ( `ETH`, `ERC20`, `NFT`) в качестве вознаграждения кому угодно, не зная адреса учетной записи получателя. Все, что вам нужно, это `GitHub`имя пользователя или `Google`адрес электронной почты получателя. Получатель может претендовать на вознаграждения, доказав право собственности на `GitHub`учетную `Google`запись (используя «Войти с помощью» `GitHub`или «Войти с помощью `Google`потока»).
> 
> Вы можете подключиться с помощью Metamask или WalletConnect. Вы также можете использовать его из своего мобильного браузера. Одной из моих целей была поддержка L1 и L2 с точки зрения разработки, развертывания и тестирования, а также обеспечение сквозной работы как с L1, так и с zkSync Era.
> 
> На данный момент проект развернут в следующих сетях:
> 
> * Основная сеть эпохи zkSync (L2)
> * Тестовая сеть zkSync Era Goerli (L2)
> * Сеполия (L1)
> * zkSync Era Local (L2) — локальный узел, вы можете использовать его, если у вас есть локальный узел и обозреватель блоков работает локально, также необходимо развернуть [контракты](https://github.com/Romsters/rewardme/blob/main/README.md#-deploying-contracts) . 
>   Поэтому обязательно используйте один из этих ^.
> 
> Никакие конфиденциальные данные нигде не хранятся, имя пользователя или адрес электронной почты хешируются перед сохранением в SC.
> 
> Порядок подачи заявки: Получите токен от Google/GitHub -> Обменяйте его на подписанный токен приложения -> Смарт-контракт проверяет токен приложения -> Смарт-контракт отправляет вам вознаграждение на выбранный вами адрес.
> 
> <img alt="Скриншот 29.10.2023, 04 03 13" width="385" src="https://private-user-images.githubusercontent.com/8478134/278847228-c0673a2f-2348-4d0e-927e-19fc7a50d346.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgxNzgzMDMsIm5iZiI6MTcwODE3ODAwMywicGF0aCI6Ii84NDc4MTM0LzI3ODg0NzIyOC1jMDY3M2EyZi0yMzQ4LTRkMGUtOTI3ZS0xOWZjN2E1MGQzNDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTdUMTM1MzIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmZmMWFlM2FiN2M4MjdmMGMwNjZlNWUyYzhlOGJiYWQ0Njg2YWRmMWFkNTZhNjlmOTgzMjQyNGNhZWNmZjZkYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wFj8igjPh7DViSqt3gskyqDbFSVxQLqxDVz2eBQPXlE"> <img alt="Скриншот 29.10.2023, 07.04.24" width="291" src="https://private-user-images.githubusercontent.com/8478134/278847320-81c6e329-6a75-4ff2-b8a6-4b7942973e8f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgxNzgzMDMsIm5iZiI6MTcwODE3ODAwMywicGF0aCI6Ii84NDc4MTM0LzI3ODg0NzMyMC04MWM2ZTMyOS02YTc1LTRmZjItYjhhNi00Yjc5NDI5NzNlOGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTdUMTM1MzIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDhiZjJlNTMzNzI3NjQzNzI0NjM4ZGVjNzMyMTlhYTRjOGQwZmRmZDAzMzkwOTFmYWI5MzlmYjc4MjM2ZDJjNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xDGSnGgidp4iVZZJFg4aYlXU9RyhUH0xGP4WHTDJWn0">
> ### Какую конкретную обратную связь вы ищете?
> Попробуйте, дайте мне знать, что вы думаете.
> 
> ### Как сообщество может помочь или внести свой вклад в ваш проект?
> Попробуйте, протестируйте, ознакомьтесь с исходным кодом и смело используйте его повторно!
> 
> ### Дополнительная информация (необязательно)
> Проект является прототипом, не используйте его или его части в производстве, если вы не уверены на 100%.

дуже подобається zkSynk низькі комісії не то що в Ефірі

**Comment by Delaw258** [comment]
the idea is cool

**Comment by FrondEnt** [comment]
works?

**Comment by velezlia** [comment]
Project Name: RewardMe

Repo Link: [RewardMe Repo](https://github.com/romsters/rewardme)

Demo Link: [RewardMe Demo](https://rewardme-app-kxknkq7kaq-lm.a.run.app/)

Technologies Used: Solidity, Rust, React, wagmi, mui, Google & GitHub API

Description:
RewardMe is a project that enables users to send crypto assets (such as ETH, ERC20 tokens, and NFTs) as rewards to individuals without knowing their account addresses. The recipient can claim the rewards by proving ownership of their GitHub or Google account. The project supports both L1 and L2 solutions, including deployment on networks like zkSync Era Mainnet, Goerli Testnet, Sepolia, and local environments.

Feedback:
We invite the community to try out RewardMe and share their thoughts on its functionality, usability, and overall experience. Your feedback will help us improve the project and make it more user-friendly and efficient.

How to Contribute:
You can contribute to RewardMe by testing the application, exploring the source code, and providing feedback or suggestions for enhancements. Feel free to reuse any parts of the project that may be useful for your own projects, but please exercise caution as RewardMe is currently a prototype and not intended for production use unless thoroughly vetted.

Additional Information:
RewardMe is a prototype, so use it with caution and avoid deploying it in production environments without proper testing and validation.

**Comment by wunrorobudaniel** [comment]
RewardMe is a revolutionary project enabling seamless crypto asset rewards distribution to individuals without prior knowledge of their account addresses. Utilizing GitHub usernames or Google account emails, users can send rewards effortlessly. Claiming rewards is streamlined through Sign in with GitHub or Google authentication. Developed with Solidity, Rust, React, and integration with Google & GitHub APIs, RewardMe supports both L1 and L2 environments, including zkSync Era Mainnet, Goerli Testnet, and Sepolia. Join the exploration and contribute to this innovative endeavor!

**Comment by sarahzitzner** [comment]
Awesome project! I'll check it out and provide feedback. Keep up the great work!

**Comment by GaodiXC** [comment]
They are nice actually

**Comment by cosmossphere** [comment]
Loved the tech and was quite an amazing experience... Hope this gets more widely adopted 

Yahoo Mail: Search, Organize, Conquer 
 
  On Mon, Mar 11, 2024 at 5:09 PM, ***@***.***> wrote:   

They are nice actually

**Comment by Josephtran102** [comment]
Great idea!

**Comment by MasterTi1** [comment]
This is great! Real attraction and reward of people who are alienated from web3 but helpful and interested

**Comment by alexonelov** [comment]
great idea
